addons/im_livechat/static/src/embed/common/autopopup_service.js

```diff
@@ -18,13 +18,15 @@ export class AutopopupService {
         this.livechatService = livechatService;
         this.ui = ui;
 
-        browser.setTimeout(async () => {
-            await Promise.all([storeService.isReady, storeService.chatHub.initPromise]);
-            if (this.allowAutoPopup) {
-                expirableStorage.setItem(AutopopupService.STORAGE_KEY, true);
-                livechatService.open();
-            }
-        }, storeService.livechat_rule?.auto_popup_timer * 1000);
+        storeService.isReady.then(() => { 
+            browser.setTimeout(async () => {
+                await Promise.all([storeService.isReady, storeService.chatHub.initPromise]);
+                if (this.allowAutoPopup) {
+                    expirableStorage.setItem(AutopopupService.STORAGE_KEY, true);
+                    livechatService.open();
+                }
+            }, storeService.livechat_rule?.auto_popup_timer * 1000);
+        });
```

Comment:
the `await Promise.all([storeService.isReady` is superfluous in this current PR code

isn't this already enough to just move:

```js
await Promise.all([storeService.isReady, storeService.chatHub.initPromise]);
```
before the setTimeout